src/Symfony/Component/Messenger/Middleware/HandleMessageMiddleware.php

```diff
@@ -97,6 +106,14 @@ public function handle(Envelope $envelope, StackInterface $stack): Envelope
             } catch (\Throwable $e) {
                 $exceptions[$handlerDescriptor->getName()] = $e;
```

Comment:
It would be great if a HandlerFailureEvent subscriber could silence an exception.

Right now, it's only possible to log them. But by always passing the exception to the HandlerFailureEvent and dispatching that, a subscriber could say "ignore this".

A use case could be to annotate `#[IgnoreException(SomeTimeoutException::class)]` on the handler. In case of an exception, a subscriber would see that the handler wants to ignore this exception, and it will silence it. 

From Messenger perspective all will be fine again. 

Related to https://github.com/symfony/symfony/pull/50980

Reply:
We could add a method to the event `ignoreException(bool $ignore)` indeed. I don't mind adding this feature, and again I'll leave it to others to decide if adding it is a good idea.

Reply:
@ro0NL @OskarStark What do you think?

Reply:
im not really convinced about `#[IgnoreException(`, i'd prefer try/catching it at the handler level which is 100% explicit

Reply:
Also fine, but then we should let a subscriber mutate the handler descriptor. See my previous comment.